examples/block-device/block-device-snapshot/main.tf

```diff
@@ -0,0 +1,114 @@
+provider "vra" {
+  url           = var.url
+  refresh_token = var.refresh_token
+  insecure = true
+}
+
+data "vra_cloud_account_vsphere" "this" {
+  name = var.cloud_account
+}
+
+data "vra_region" "this" {
+  cloud_account_id = data.vra_cloud_account_vsphere.this.id
+  region           = var.region
+}
+
+data "vra_project" "this" {
+  name = var.project
+}
+
+resource "vra_flavor_profile" "this" {
+  name        = "tf-vra-flavor-profile"
+  description = "my flavor"
+  region_id   = data.vra_region.this.id
+
+  flavor_mapping {
+    name          = "small"
+    cpu_count = "1"
+    memory = 1024
+
+  }
+
+  flavor_mapping {
+    name          = "medium"
+    cpu_count = "2"
+    memory = 2048
+  }
+}
+
+resource "vra_image_profile" "this" {
+  name        = "tf-vra-image-profile"
+  description = "terraform test image profile"
+  region_id   = data.vra_region.this.id
+
+  image_mapping {
+    name       = "ubuntu"
+    image_name = var.image_name
+  }
+}
+
+resource "vra_block_device" "disk1" {
+  capacity_in_gb = 10
+  name = "terraform_vra_block_device1"
+  project_id = data.vra_project.this.id
+  persistent = true
+
+  constraints {
+    mandatory  = true
+    expression = "foo:bar"
+  }
+}
```

Comment:
Remove only the `vra_flavor_profile`, and `vra_image_profile` resources.

Also add `depends_on = [vra_storage_profile.this]` in both `vra_block_device.disk1` and `vra_block_device.disk2` so that Storage Profile is created first before the disks are created.

Reply:
Done